src/tools/cephfs/first-damage.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+# Ceph - scalable distributed file system
+#
+# Copyright (C) 2022 Red Hat, Inc.
+#
+# This is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 2.1, as published by the Free Software
+# Foundation.  See file COPYING.
+
+# Suggested recovery sequence (for single MDS cluster):
+#
+# 1) Unmount all clients.
+#
+# 2) Flush the journal (if possible):
+#
+#    ceph tell mds.<fs_name>:0
```

Comment:
This seems incomplete.

Also, AFAICT flushing the journal basically doesn’t do anything, so we can probably stop recommending this. (It specifically *does not* push metadata out to backing omap storage, it just issues a Journaler flush, which is quite boring.)

Reply:
I did forget `flush journal`.

It actually does do a full flush of metadata to backing omap.

Reply:
Weird. I've looked at this before and (with concern!) come to the opposite conclusion, but indeed there is C_Flush_Journal starting a new segment and then doing trim_mdlog()...